### ISSUE TITLE
Visual automation not working - pending checks

### ISSUE THREAD

**Issue by jamesnyika** [kelaberetiv/TagUI#1182]
Hi 
 Every attempt i have made to use the click *.png image does not work. It says it cannot find the image
 The docs are quite scant on information about this feature - what do I need to have installed to make it work. I am on a mac and I suspect it needs SikuliX ? am I right or not.. ? if it does - what is the correct procedure to install SikuliX on a mac ?

**Comment by ruthtxh** [comment]
Hi @jamesnyika, the installation comes together when you installed TagUI. As long as you follow the installation instructions [here](https://tagui.readthedocs.io/en/latest/setup.html?#macos-linux) it should work fine.

Can you share the exact error message you are getting? For visual automation to work, it is best that the browser zoom setting is at 100%. To set this on the TagUI chrome user profile, you can run the command `tagui live` in the terminal and adjust the zoom level of that browser.

**Comment by jamesnyika** [comment]
Hi @ruthtxh : I noticed on the install instructions, there is a reference to the corretto install of OpenJDK:` amazon-corretto-8.322.06.1-macosx-x64.pkg `specifically. Do I need to use that specific one or can I use ANY OpenJDK install ? 

 Mine is : 

```
$> java -version
openjdk version "17.0.2" 2022-01-18
OpenJDK Runtime Environment Homebrew (build 17.0.2+0)
OpenJDK 64-Bit Server VM Homebrew (build 17.0.2+0, mixed mode, sharing)
```

I have a valid png which is a picture of the button I want to click. Browser is zoomed to 100%. 

The message I get is 

 ```
START - automation started - Mon Feb 14 2022 16:24:50 GMT-0500 (EST)
LIVE MODE - type done to quit
https://my.site.com
click newjob.png
ERROR - cannot find /Users/jn/experimental/tags/newjob.png on screen
click ./newjob.png
ERROR - cannot find /Users/jn/experimental/tags/./newjob.png on screen
```

**Comment by ruthtxh** [comment]
Hi @jamesnyika as long as the openJDK is 64bit it should work fine.

Since you're on MacOS, a known issue is that the retina display screenshots are of a resolution twice as high as the resolution sikuli uses.  You may check the resolution of your captured image, whether it is 144dpi or 72dpi.

One way is to downsize this manually (cmd+shift+4) or to add this vision step at the start of your workflow so that it will automatically adjust the resolution:
```
vision Settings.AlwaysResize = 0.5
```

**Comment by jamesnyika** [comment]
Hi @ruthtxh 
 Well.. some good news. I was able to make it work but it would save a lot of people time if we can add this info to the docs. 
 
 On Mac
   1. You need to be careful which terminal you run your tagui command line from. I ran mine from inside VSCode and so you will get (but might miss) a dialog that requires you authorize VS code to be able to record your screen. Once this is accepted, you must restart VS Code and then try re-running your automation again. Looks like it will work. If you use a straight Mac terminal or iTErm (in my case), then you will need to perform the same authorization step in the Mac Settings / Security area to allow recording of the screen. 

 2. The suggested setting for vision : ```vision Settings.AlwaysResize = 0.5  ``` at the top of your tag file is key. It allows you to use the higher res Retina images without doing your own manual downsizing. Alternatively, you can change their DPI setting down from 144 to 72 (I used Mac Preview to do it under the Adjust Size menu). 

3. **Very important** - the browser resolution when you take your png picture that will be used by Sikuli MUST BE THE SAME as the browser resolution of the page when it runs the automation. For example, if you take a picture for the page at 75% resolution (browser resolution that is - not your display) and then try to click on that image when the automation is running but it opens a browser at 100% resolution - then Sikuli will NOT find your button or image.
4. One nice thing is that if your button is large and long, you do not need to capture it exactly - you can just capture the area that makes it unique (eg. the text) and that too will work. 

Thank you SO SO Much @ruthtxh  for your help on this. I appreciate what you did for me here. 
I hope this helps anyone else looking into this issue.

**Comment by ruthtxh** [comment]
Thanks for sharing back the details @jamesnyika, I will review the documentation and update accordingly.

**Comment by Steve2955** [comment]
Just to document another possible point of failure if someone happens to run into issues with running visual automations:

I just installed tagui v4.48 using the installer and wrote a basic flow. Everything would work as expected except the visual automation (e.g. `click searchbar.png`). When reaching the visual automation step, tagui would just wait endlessly. Nothing happened with no error message whatsoever.

Turns out there was some kind of issue with my python installation. After reinstalling python the flow executed correctly.
